src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php

```diff
@@ -150,14 +150,21 @@ public function register_routes() {
 	 * Returns the fallback template for the given slug.
 	 *
 	 * @since 6.1.0
+	 * @since 6.3.0 Ignore empty templates.
 	 *
 	 * @param WP_REST_Request $request The request instance.
 	 * @return WP_REST_Response|WP_Error
 	 */
 	public function get_template_fallback( $request ) {
 		$hierarchy         = get_template_hierarchy( $request['slug'], $request['is_custom'], $request['template_prefix'] );
-		$fallback_template = resolve_block_template( $request['slug'], $hierarchy, '' );
-		$response          = $this->prepare_item_for_response( $fallback_template, $request );
+
+		do {
```

Comment:
Instead of a while loop, why not just use a for each loop and break / continue?

Reply:
Thanks for reviewing. Good question. I don't know the original intention behind the change 🤷 I just backported.

Reply:
I can only speculate @jorgefilipecosta wanted it to run at least once to set `$fallback_template`, but he's not around.

Reply:
It's just code style personally, I prefer to avoid continue, break as much as needed I'm guessing the idea is to stop when a fallback is found.